##### Задача 1

Известно, что генеральная совокупность распределена нормально 
со средним квадратическим отклонением $\sigma=5$. 

Найти доверительный интервал для оценки математического ожидания $a$ с надежностью 0.95, 
если выборочная  средняя M = 24.15, а объем выборки n = 100.

$\delta = \frac{t_{\lambda} \cdot \sigma}{\sqrt{n}}$

2Ф (t\_lambda) = lambda_

In [28]:
lambda_ = 0.95

Ф (t\_lambda) = 0.475

In [29]:
t_lambda = 1.96

In [30]:
delta = 1.96 * 5 / 10

In [31]:
delta

0.9800000000000001

Доверительный интервал:

(24.15 - delta, 24.15 + delta)

In [32]:
24.15 - delta

23.169999999999998

In [33]:
24.15 + delta

25.13

Ответ: (23.17, 25.13).

Таблица значений функции Лапласа:

<img src='https://mathprofi.net/files/u/laplas.png'>Таблица значений функции Лапласа</img>

##### Задача 2

In [1]:
import numpy as np

В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, 
полученные опытные данные:
7.1, 6.3, 6.2, 5.8, 7.7, 6.8, 6.7, 5.9, 5.7, 5.1
Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, 
оценить истинное значение величины X при помощи доверительного интервала, покрывающего это 
значение с доверительной вероятностью 0,95.

In [2]:
a = np.array([7.1, 6.3, 6.2, 5.8, 7.7, 6.8, 6.7, 5.9, 5.7, 5.1])

In [3]:
a

array([7.1, 6.3, 6.2, 5.8, 7.7, 6.8, 6.7, 5.9, 5.7, 5.1])

In [4]:
a_mean = a.mean()

In [5]:
a_mean

6.33

In [9]:
a_std = (np.sum((a - a_mean)**2) / 10)**0.5
a_std

0.722564875980005

In [10]:
a_std = (np.sum((a - a_mean)**2) / (10-1))**0.5
a_std

0.7616502551113018

In [8]:
a.std()

0.722564875980005

In [12]:
a_std = a.std(ddof=1)
a_std

0.7616502551113018

In [13]:
t_kr = 2.262

$\delta = \frac{t_{\lambda} \cdot \sigma}{\sqrt{n}}$

In [23]:
delta = t_kr * a_std / (10**0.5)

In [24]:
delta

0.5448139164889237

In [25]:
a_mean - delta

5.785186083511077

In [26]:
a_mean + delta

6.8748139164889235

ДИ = (5.785, 6.8748)

<img src='http://900igr.net/up/datas/109946/035.jpg'>Таблица критических точек t-распределения Стьюдента</img>

##### Задача 3

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 10 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=16 шариков средний диаметр оказался равным 10,3 мм, а дисперсия известна и равна 1 кв.мм.

H0: M = 10

H1: M > 10

$\delta = \frac{t_{\lambda} \cdot \sigma}{\sqrt{n}}$

Выведем нахождение значение критерия из предыдущей формулы (наблюдаемый критерий обозначим как U_nabl)

U_nabl = delta \* (n\*\*0.5) / sigma

In [38]:
U_nabl = (10.3 - 10) * (16**0.5) / 1
U_nabl

1.2000000000000028

Найдем критическую точку для односторонней критической области (по таблице Лапласа):

In [39]:
alpha = 0.05

In [40]:
Ф = (1-2*alpha) / 2

In [41]:
Ф

0.45

In [42]:
u_kr = 1.645

In [43]:
d = 10

In [44]:
U_nabl < u_kr

True

Ответ: принимаем H0.

##### Задача 4

Продавец утверждает, что средний вес пачки чая составляет 100 г. Из партии извлечена выборка и взвешена. Вес каждой пачки - см. значения массива p_ch. Не противоречит ли это утверждению продавца? Доверительная вероятность 99%. Вес пачек чая распределен нормально.

In [47]:
p_ch = [101, 102, 99, 98, 103, 102, 99, 97]

Задача решается аналогично предыдущей, но для поиска критических значений критерия используется 
не таблица значений функции Лапласа, а таблица критических точек t-распределения Стьюдента, так как неизвестна дисперсия генеральной совокупности. При этом несмещенную оценку дисперсию нужно найти по имеющейся выборке.